### DISCUSSION TITLE
[IOS][DRM] generate custom assetIdData to makeStreamingContentKeyRequestData

### DISCUSSION THREAD

**Discussion by matthieu-renard-wiztivi** [TheWidlarzGroup/react-native-video#4312] [Q&A]
### What are you trying to do?

For a client, I'm integrating a Fairplay DRM stream but an error is triggered on the `makeStreamingContentKeyRequestData` step.

To fix my issue I need to change the way `assetIdData` is created in `DRMManager.swift` file, at the line [L129](https://github.com/TheWidlarzGroup/react-native-video/blob/569a79c510127874eedf9f4c1963f87df3db620d/ios/Video/Features/DRMManager.swift#L129).

My questions are:

- Can I change this behavior to avoid local modification of node modules by overriding the DRMManager class on project side in the ios folder?
- After checking the code, I didn't see a javascript hook to override this part like `getLicense`, did you plan to have this feature?

I imagine a feature allowing to return `assetIdData` according to the contentId and the keyrequest from a javascript hook if defined, and default behavior if not. Like the `getLicense` hook.

### What have you tried?

If I apply my custom modification to build `assetIdData`, the `makeStreamingContentKeyRequestData` method succeeds, then I use the javascript hook `getLicense`, and it works perfectly, I'm able to stream the videos.

### Platforms

iOS

**Comment by freeboub** [comment]
Old discussion sorry for the late reply...
Did you try with the last release of the package, there was some major changes in 6.6.2 : https://github.com/TheWidlarzGroup/react-native-video/pull/4136
If it is still not OK, I think you should modify or add hook function to modify the problematic ID

**Reply by matthieu-renard-wiztivi** [reply]
Hello @freeboub ,

Thank you for your answer.

Yes I tested with the release version 6.7.0.

Did you have a mechanism allowing hook like this or did you plan to do it in futur release ? 
I didn't check the code since the 6.7.0 but with this version I wasn't able to do it with a hook, but able to do it by local changes.

Regards,
Matthieu.
